src/ssl.ml

```diff
@@ -301,26 +306,137 @@ end
 
 (* Same as above, but doesn't release the lock. *)
 module Runtime_lock_base = struct
-  external connect : socket -> unit = "ocaml_ssl_connect"
-
-  external accept : socket -> unit = "ocaml_ssl_accept_blocking"
-
-  external write : socket -> Bytes.t -> int -> int -> int
-    = "ocaml_ssl_write_blocking"
-  external write_substring : socket -> string -> int -> int -> int
-    = "ocaml_ssl_write_blocking"
-  external write_bigarray : socket -> bigarray -> int -> int -> int
-    = "ocaml_ssl_write_bigarray_blocking"
-
-  external read : socket -> Bytes.t -> int -> int -> int
-    = "ocaml_ssl_read_blocking"
-
-  external read_into_bigarray :
-    socket -> bigarray -> int -> int -> int = "ocaml_ssl_read_into_bigarray_blocking"
-
-  external flush : socket -> unit = "ocaml_ssl_flush_blocking"
-
-  external ssl_shutdown : socket -> bool = "ocaml_ssl_shutdown_blocking"
+  external raw_connect : socket -> int = "ocaml_ssl_connect_blocking"
+
+  let connect socket =
+    let ret = raw_connect socket in
+    let err = get_error socket ret in
+    if err != Error_none then
+      raise (Connection_error err)
+
+  external raw_accept : socket -> int = "ocaml_ssl_accept_blocking"
+
+  let accept socket =
+    let ret = raw_accept socket in
+    let err = get_error socket ret in
+    if err != Error_none then
+      raise (Accept_error err)
+
+  external raw_write : socket -> Bytes.t -> int -> int -> int
+    = "ocaml_ssl_write_blocking" [@@noalloc]
+
+  let write socket buffer start length =
```

Comment:
unrelated thought: The new `Runtime_lock` module makes me want to write a `Labels` module where `start` and `length` are `~off ~len` arguments.

Reply:
Good idea.

Reply:
Maybe another PR ?

Reply:
sure, but low priority.